draft-ietf-quic-invariants.md

```diff
@@ -150,10 +150,25 @@ version-specific semantics are marked with an X.
 {: #fig-long title="QUIC Long Header"}
 
 A QUIC packet with a long header has the high bit of the first octet set to 1.
+All other bits in that octet are version specific.
+
+The next octet contains the length in octets of the two Connection IDs (see
+{{connection-id}}) that follow.  Each length is encoded as a 4 bit unsigned
+integer.  The length of the Destination Connection ID occupies the high bits of
+the octet and the length of the Source Connection ID occupying the low bits of
+the octet.  An encoded length of 0 indicates that the connection ID is also 0
+octets in length.  Non-zero encoded lengths are increased by 7 to get the full
+length of the connection ID.  For example, an octet with the value 0xa0
+describes a 17 octet Destination Connection ID and a zero octet Source
+Connection ID.
 
-A QUIC packet with a long header has two fixed fields immediately following the
-first octet: a 64-bit Connection ID (see {{connection-id}}) and a 32-bit Version
-(see {{version}}).
+The connection ID lengths are followed by connection IDs.  The connection ID
+selected by the recipient of the packet (the Destination Connection ID) is
+followed by the connection ID selected by the sender of the packet (the Source
+Connection ID).
+
+After both Connection IDs, a 32-bit Version (see {{version}}) is followed by a
+version-specific payload.
```

Comment:
Not in this PR, but I think moving to varlen connection IDs motivates moving the version in front of the connection IDs, because then it's in a fixed spot, which is nice.

Previously the connection ID was always in a fixed spot, but now it's offset changes by 1 byte based on long/short header anyway, so we might as well have it move by 5 bytes instead of 1.  Conveniently, an 8 byte connection ID fits into a typical 16 byte cacheline in either location.

This isn't to say that the connection ID format in long headers should be removed from the invariants, because I think that turns out to be impractical.

Reply:
Yep, I think that's a good idea.  We should make that move sooner rather than later if we do.